qiskit/transpiler/passes/utils/gates_basis.py

```diff
@@ -18,22 +18,37 @@
 class GatesInBasis(AnalysisPass):
     """Check if all gates in a DAG are in a given set of gates"""
 
-    def __init__(self, basis_gates):
+    def __init__(self, basis_gates, target=None):
         """Initialize the GatesInBasis pass.
 
         Args:
             basis_gates (list): The list of strings representing the set of basis gates.
+            target (Target): If specified the target represnting the backend. If specified
+                this will be used instead of the ``basis_gates`` parameter
         """
         super().__init__()
         self._basis_gates = set(basis_gates)
+        self._target = target
```

Comment:
It might be good to give `basis_gates` a default value, so callers don't need to provide nonsense data if they only want to pass the `target`.  That's a consistent signature with the previous one, so I think it's still ok for a backport?

Reply:
Normally I agree that would be a good idea. But, if we want to use this pass in the preset passmanagers making the options mutually exclusive will make our lives a bit more difficult. This is mostly because for backwards compat the preset passmanagers will have both the `basis_gates` list (which is pulled from the target) and `target` defined with a `BackendV2` backend. But for `BackendV1` only `basis_gates` is defined. So if we made this mutually exclusive we'd have to have an if/else block around the backend version for the constructor of this pass. I think it's probably just easier to document that target will take precedence over the other argument so we can have a single constructor in the preset passmanagers regardless of backend version.

I will change this to have a default though and will raise an error if neither is set.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7548/commits/c55d4697d2e1ee26df4ddaa14916528a66ff33b1